codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/protocol/MessageFrame.java

```diff
@@ -0,0 +1,148 @@
+    /**
+     * Creates a new instance with the specified messages.
+     *
+     * @param messages the messages that this message frame will contain.
+     */
+    public MessageFrame(final String... messages) {
+        this(new ArrayList<String>(Arrays.asList(messages)));
+    }
+
+    /**
+     * Creates a new instance with the specified messages.
+     *
+     * @param messages the messages that this message frame will contain.
+     */
+    public MessageFrame(final List<String> messages) {
+        super(generateContent(messages));
+        this.messages = messages;
+    }
+
+    /**
+     * Returns this {@link MessageFrame}s messages.
+     *
+     * @return {@code List} the messages for this {@link MessageFrame}
+     */
+    public List<String> messages() {
+        return Collections.unmodifiableList(messages);
+    }
+
+    @Override
+    public MessageFrame copy() {
+        return new MessageFrame(messages);
+    }
+
+    @Override
+    public MessageFrame duplicate() {
+        return new MessageFrame(messages);
+    }
+
+    @Override
+    public MessageFrame retain() {
+        super.retain();
+        return this;
+    }
+
+    @Override
+    public MessageFrame retain(int increment) {
+        super.retain(increment);
+        return this;
+    }
+
+    @Override
+    public String toString() {
+        return StringUtil.simpleClassName(this) + "[messages=" + messages + ']';
+    }
+
+    private static ByteBuf generateContent(final List<String> messages) {
+        final JsonStringEncoder jsonEndocder = new JsonStringEncoder();
+        final ByteBuf content = Unpooled.buffer();
+        content.writeByte('a').writeByte('[');
+        final int size = messages.size();
+        for (int i = 0; i < size; i++) {
+            content.writeByte('"');
+            final String element = messages.get(i);
+            if (element == null) {
+                messages.subList(i, size).clear();
+                break;
+            }
+            final String escaped = escapeCharacters(jsonEndocder.quoteAsString(element));
+            final ByteBuf escapedBuf = Unpooled.copiedBuffer(escaped, CharsetUtil.UTF_8);
+            content.writeBytes(escapedBuf).writeByte('"');
+            escapedBuf.release();
+            if (i < size - 1) {
+                content.writeByte(',');
+            }
+        }
+        return content.writeByte(']');
+    }
+
+    private static String escapeCharacters(final char[] value) {
+        final StringBuilder sb = new StringBuilder(value.length);
+        for (char ch : value) {
+            if (ch >= '\u0000' && ch <= '\u001F' ||
+                    ch >= '\uD800' && ch <= '\uDFFF' ||
+                    ch >= '\u200C' && ch <= '\u200F' ||
+                    ch >= '\u2028' && ch <= '\u202F' ||
+                    ch >= '\u2060' && ch <= '\u206F' ||
+                    ch >= '\uFFF0' && ch <= '\uFFFF') {
+                final String ss = Integer.toHexString(ch);
+                sb.append('\\');
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) StringBuffer (or StringBuilder).append is called 2 consecutive times with literal Strings. Use a single append with a single combined String. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AConsecutiveLiteralAppends)